src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -465,6 +466,16 @@ File {} line {}.)", wname, location.keyword, location.filename, location.lineno)
             this->applyEXIT(handlerContext.keyword, handlerContext.currentStep);
     }
 
+    void Schedule::handleFBHPDEF(HandlerContext& handlerContext)
+    {
+        using FBHP = ParserKeywords::FBHPDEF;
+        const auto& record = handlerContext.keyword.getRecord(0);
+        ScheduleState::BHPDefaults bhp_defaults;
+        bhp_defaults.prod_target = record.getItem<FBHP::TARGET_BHP>().getSIDouble(0);
+        bhp_defaults.inj_limit = record.getItem<FBHP::LIMIT_BHP>().getSIDouble(0);
```

Comment:
You should probably guard the `.getSIDouble(0)` access with something like
```C++
if (const auto& limit = record.getItem<FBHP::TARGET_BHP>(); ! limit.defaultApplied(0)) {
    bhp_defaults.prod_target = limit.getSIDouble(0);
}
if (const auto& limit = record.getItem<FBHP::LIMIT_BHP>(); ! limit.defaultApplied(0)) {
    bhp_defaults.inj_limit = limit.getSIDouble(0);
}
```
That way, `nullopt` will signify "there is no field-wide default BHP limit for this well type".

Reply:
Problem with that is that it clearly contradicts the documentation for FBHPDEF where it explictly says that either than be defaulted. It's tricky to get this correct with all the constraints of documentation not matching behavior, but I do not think we should directly break this explicit statement in the documentation ?

Reply:
Okay, but we also need to handle the case of both items being defaulted at the same time which should be treated as if the keyword was not specified at all.  Maybe something like
```C++
const auto& prod_limit = record.getItem<FBHP::TARGET_BHP>();
const auto& inj_limit = record.getItem<FBHP::LIMIT_BHP>();
if (! (prod_limit.defaultApplied(0) && inj_limit.defaultApplied(0))) {
    bhp_defaults.prod_target = prod_limit.getSIDouble(0);
    bhp_defaults.inj_limit = inj_limit.getSIDouble(0);
}
this->snapshots.back().bhp_defaults.update(std::move(bhp_defaults));
```
instead?

Reply:
ok.